### DISCUSSION TITLE
Wezterm doesn't recognize the bold variant of a custom Iosevka font

### DISCUSSION THREAD

**Discussion by crides** [wezterm/wezterm#5202] [Q&A]
### What Operating System(s) are you running on?

Linux X11, macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

awesomewm

### WezTerm version

20240102-011725-2f7ca41a

### Ask your question!

I have a custom Iosevka font `Iosevka Term Thin`, and wezterm doesn't seem to recognize the bold variant. Here's what it looks like (the bold portions are just fallback fonts):
![maim-2024 03 21 21 18 34](https://github.com/wez/wezterm/assets/15631555/cbbf292e-65b5-493a-a1b4-13b73f67856d)

Wezterm does recognize the font's _existance_:
```
$ wezterm ls-fonts --list-system
... snip ...
wezterm.font("Iosevka Term Thin", {weight="Thin", stretch="Normal", style="Normal"}) -- /home/steven/.local/share/fonts/Iosevka-Term-Thin-Regular.ttf, FontConfig
wezterm.font("Iosevka Term Thin", {weight="Thin", stretch="Normal", style="Italic"}) -- /home/steven/.local/share/fonts/Iosevka-Term-Thin-Italic.ttf, FontConfig
wezterm.font("Iosevka Term Thin", {weight="Thin", stretch="Normal", style="Oblique"}) -- /home/steven/.local/share/fonts/Iosevka-Term-Thin-Oblique.ttf, FontConfig
wezterm.font("Iosevka Term Thin", {weight="Bold", stretch="Normal", style="Normal"}) -- /home/steven/.local/share/fonts/Iosevka-Term-Thin-Bold.ttf, FontConfig
wezterm.font("Iosevka Term Thin", {weight="Bold", stretch="Normal", style="Italic"}) -- /home/steven/.local/share/fonts/Iosevka-Term-Thin-BoldItalic.ttf, FontConfig
wezterm.font("Iosevka Term Thin", {weight="Bold", stretch="Normal", style="Oblique"}) -- (AKA: Iosevka Term Thin Oblique) /home/steven/.local/share/fonts/Iosevka-Term-Thin-BoldOblique.ttf, FontConfig
... snip ...
```
But somehow doesn't recognize it for using:
```
$ wezterm ls-fonts
Primary font:
wezterm.font_with_fallback({
  -- /home/steven/.local/share/fonts/Iosevka-Term-Thin-Regular.ttf, FontConfig
  {family="Iosevka Term Thin", weight="Thin"},

  -- /home/steven/.local/share/fonts/JetBrainsMono-Regular.ttf, FontConfig
  "JetBrains Mono",

  -- /usr/share/fonts/noto/NotoColorEmoji.ttf, FontConfig
  -- Assumed to have Emoji Presentation
  -- Pixel sizes: [128]
  "Noto Color Emoji",

  -- <built-in>, BuiltIn
  "Symbols Nerd Font Mono",

})
... snip ...
When Intensity=Bold Italic=false:
wezterm.font_with_fallback({
  -- /home/steven/.local/share/fonts/JetBrainsMono-Regular.ttf, FontConfig
  "JetBrains Mono",

  -- /usr/share/fonts/noto/NotoColorEmoji.ttf, FontConfig
  -- Assumed to have Emoji Presentation
  -- Pixel sizes: [128]
  "Noto Color Emoji",

  -- <built-in>, BuiltIn
  "Symbols Nerd Font Mono",

})
```

Since this is Iosevka, I have this font's weights configured to be a bit different:
```toml
[buildPlans.iosevka-term.weights.Regular]
shape = 300
menu = 400
css = 300

[buildPlans.iosevka-term.weights.Bold]
shape = 500
menu = 700
css = 600
```
where the `shape` is the font weight actually used for the variant, and `menu` is (I think) what the weight is represented as to the application.

What could be the problem here?

**Comment by wez** [comment]
wezterm will use the weight of the base font + 200 to find the bold font variant.
I think in this case, since your base font is `Thin` (100), the bolder weight is computed to be `Light` (300).  When matching, the closest available weight is still `Thin`.

If you're going to be using funky weights like this, then I recommend using https://wezfurlong.org/wezterm/config/lua/config/font_rules.html to explicitly set the font for the different styles.
